Tools/jit/build.py

```diff
@@ -0,0 +1,695 @@
+                s = s.removeprefix(self.prefix)
+                value, symbol = HoleValue.GOT, s
+                addend = 0
+            case {
+                "Type": {"Value": kind},
+                "Section": {"Value": s},
+                "Offset": offset,
+            } | {
+                "Type": {"Value": kind},
+                "Symbol": {"Value": s},
+                "Offset": offset,
+            }:
+                offset += base
+                s = s.removeprefix(self.prefix)
+                value, symbol = _symbol_to_value(s)
+                addend = 0
+            case _:
+                raise NotImplementedError(relocation)
+        # Turn Clang's weird __bzero calls into normal bzero calls:
+        if symbol == "__bzero":
+            symbol = "bzero"
+        return Hole(offset, kind, value, symbol, addend)
+
+
+def get_target(
+    host: str, *, debug: bool = False, verbose: bool = False
+) -> COFF | ELF | MachO:
+    target: COFF | ELF | MachO
+    if re.fullmatch(r"aarch64-apple-darwin.*", host):
+        target = MachO("aarch64-apple-darwin", alignment=8, prefix="_")
+    elif re.fullmatch(r"aarch64-.*-linux-gnu", host):
+        target = ELF("aarch64-unknown-linux-gnu", alignment=8)
+    elif re.fullmatch(r"i686-pc-windows-msvc", host):
+        target = COFF("i686-pc-windows-msvc", prefix="_")
+    elif re.fullmatch(r"x86_64-apple-darwin.*", host):
+        target = MachO("x86_64-apple-darwin", prefix="_")
+    elif re.fullmatch(r"x86_64-pc-windows-msvc", host):
+        target = COFF("x86_64-pc-windows-msvc")
+    elif re.fullmatch(r"x86_64-.*-linux-gnu", host):
+        target = ELF("x86_64-unknown-linux-gnu")
+    else:
+        raise ValueError(host)
+    return dataclasses.replace(target, debug=debug, verbose=verbose)
+
+
+def dump_header() -> typing.Generator[str, None, None]:
+    yield f"// $ {shlex.join([sys.executable, *sys.argv])}"
+    yield ""
+    yield "typedef enum {"
+    for kind in sorted(typing.get_args(schema.HoleKind)):
+        yield f"    HoleKind_{kind},"
+    yield "} HoleKind;"
+    yield ""
+    yield "typedef enum {"
+    for value in sorted(HoleValue, key=lambda value: value.name):
+        yield f"    HoleValue_{value.name},"
+    yield "} HoleValue;"
+    yield ""
+    yield "typedef struct {"
+    yield "    const uint64_t offset;"
+    yield "    const HoleKind kind;"
+    yield "    const HoleValue value;"
+    yield "    const void *symbol;"
+    yield "    const uint64_t addend;"
+    yield "} Hole;"
+    yield ""
+    yield "typedef struct {"
+    yield "    const size_t body_size;"
+    yield "    const unsigned char * const body;"
+    yield "    const size_t holes_size;"
+    yield "    const Hole * const holes;"
+    yield "} Stencil;"
+    yield ""
+    yield "typedef struct {"
+    yield "    const Stencil code;"
+    yield "    const Stencil data;"
+    yield "} StencilGroup;"
+    yield ""
+
+
+def dump_footer(opnames: list[str]) -> typing.Generator[str, None, None]:
```

Comment:
```suggestion
def dump_footer(opnames: list[str]) -> typing.Iterator[str]:
```

Reply:
`Iterable` means that it's can be reused. I think here should be `Iterator[str]`. Same for `dump` function

Reply:
Both are fine :)

(but, mypy does not make this assumption about reusability: https://mypy-play.net/?mypy=latest&python=3.12&flags=strict&gist=c8d554830f0ab2731f4072c4ec6cb489)